packages/devtools_app/lib/src/memory/flutter/memory_heap_tree_view.dart

```diff
@@ -83,22 +84,39 @@ class HeapTreeViewState extends State<HeapTree> with AutoDisposeMixin {
     addAutoDisposeListener(controller.leafSelectedNotifier, () {
       setState(() {});
     });
+
+    addAutoDisposeListener(controller.searchNotifier, () {
+      if (escapeHit) {
+        setState(() {
+          escapeHit = false;
+        });
+      }
+    });
   }
 
   @override
   void dispose() {
+    // Clean up the controller when the widget is removed from the
+    // widget tree.
+    textFieldController.dispose();
     super.dispose();
   }
 
   @override
   Widget build(BuildContext context) {
-    // TODO(terry): Make HeatMapSizeAnalyzer a stateful widget.
-    heapGraph = controller.theGraph;
-    snapshotDisplay = controller.snapshotByLibraryData == null
-        ? const Text('')
-        : controller.showHeatMap
-            ? HeatMapSizeAnalyzer()
-            : MemorySnapshotTable();
+    if (controller.snapshotByLibraryData != null) {
+      if (controller.showHeatMap) {
+        snapshotDisplay = HeatMapSizeAnalyzer(
+          child: SizedBox.expand(
+            child: FlameChart(controller),
+          ),
+        );
+      } else {
+        snapshotDisplay = MemorySnapshotTable();
+      }
+    } else {
+      snapshotDisplay = const Text('');
```

Comment:
nit: use `const SizedBox()` instead of `const Text('')`

Reply:
Done.